src/main/scala/com/fulcrumgenomics/util/ProgressLogger.scala

```diff
@@ -42,9 +43,48 @@ case class ProgressLogger(logger: Logger,
   /** Method to use to record progress when genome location isn't available or relevant. */
   def record(): Boolean = record(null, -1)
 
-  def record(rec: SamRecord): Boolean = record(rec.asSam)
+  def record(rec: SamRecord): Boolean = super.record(rec.asSam)
+
+  def record(variant: Variant): Boolean = record(variant.chrom, variant.pos)
+
+  def record(interval: Interval): Boolean = record(interval.getContig, interval.getStart)
+
+  /** Record progress of a single item.
+    *
+    * @param item the item log
+    * @param fallback the fallback method to record progress if the type is not a type supported by [[ProgressLogger]].
+    * @return
+    */
+  def record[ItemType](item: ItemType,
+                       fallback: (ProgressLogger, ItemType) => Boolean = (p: ProgressLogger, _: ItemType) => p.record()): Boolean = {
+    item match {
+      case rec: SamRecord            => record(rec)
+      case (chrom: String, pos: Int) => record(chrom, pos)
+      case variant: Variant          => record(variant)
+      case interval: Interval        => record(interval)
+      case _                         => fallback(this, item)
+    }
+  }
 
   /** Logs the last record if it wasn't already logged. */
   def logLast(): Boolean = super.log() // Calls the super's log() method, not log(message: String*)
+}
+
+object ProgressLogger {
 
+  /** An implicit class that wraps an [[Iterator]] and provides a method to record progress as items are consumed. */
+  implicit class ProgressLoggingIterator[ItemType](iterator: Iterator[ItemType]) {
```

Comment:
I feel like there's probably a good way, using implicits, to limit the use of the class to iterators over types that can actually be logged.